tensorboard/data/server/run.rs

```diff
@@ -161,12 +188,15 @@ impl StageTimeSeries {
 const COMMIT_INTERVAL: Duration = Duration::from_secs(5);
 
 impl<R: Read> RunLoader<R> {
-    pub fn new(run: Run) -> Self {
+    pub fn new(run: Run, plugin_sampling_hint: Arc<PluginSamplingHint>) -> Self {
         Self {
             run,
             files: BTreeMap::new(),
             checksum: true,
-            data: RunLoaderData::default(),
+            data: RunLoaderData {
+                plugin_sampling_hint,
+                ..RunLoaderData::default()
```

Comment:
Do you think that it would make sense to remove the `Default`
implementation and instead provide `RunLoaderData::new` that requires
an `Arc<PluginSamplingHint>`? No strong preference, and happy to
defer to you; just curious for your thoughts.